woocommerce/payment-gateway/class-sv-wc-payment-gateway-payment-form.php

```diff
@@ -1013,7 +1032,27 @@ public function render_fieldset_end() {
 	 */
 	public function render_js() {
 
-		wc_enqueue_js( $this->get_safe_handler_js() );
+		// skip if script already rendered
+		if ( true === $this->js_rendered ) {
+			return;
+		}
+
+		$safe_handler_js = '<script type="text/javascript">jQuery(function($) { ' . $this->get_safe_handler_js() . ' });</script>';
+
+		// mark the script as rendered
+		$this->js_rendered = true;
+
+		/**
+		 * Payment Gateway Payment Form Safe Handler Javascript.
+		 *
+		 * Filters the js code to handler and initialize the payment form.
+		 *
+		 * @since 5.10.7
+		 *
+		 * @param string $safe_handler_js
+		 * @param SV_WC_Payment_Gateway_Payment_Form $this payment form instance
+		 */
+		echo apply_filters( 'wc_' . $this->get_gateway()->get_id() . '_payment_form_safe_handler_js', $safe_handler_js, $this );
```

Comment:
why this method returns a filter? again I am not clear of the purpose

Reply:
Yeah it's a new filter I introduced as before the JS didn't have a way for 3rd party to integrate with

Reply:
let's not introduce new things if there's no reason for, or it is asked in the implementation

if you introduce something in the public API it means we have to maintain it - unless the filter is part of the fix, that our plugins need to implement I wouldn't suggest to add it in this case